returnn/hdf.py

```diff
@@ -320,7 +318,8 @@ def tasks(self):
         yield Task("run", rqmt=self.rqmt, args=range(1, (len(self.alignment_caches) + 1)))
 
     def _write_text(self, path, segment_list):
-        with open(path, "w") as f:
+        write_flag = "a" if os.path.isfile(path) else "w"
```

Comment:
I am pretty certain there will be issues with race conditions when parallel jobs append to the same file at the same time.

Reply:
Sorry, when you criticize you should always give constructive critique and show ways for improvements:

One way around the race conditions would be to still write into single files for each subtask (but those are not outputs and can be omitted if no files are included), and then add one additional task that collects all individual files into the output file which always exists, even if empty.

Reply:
>  I do not know how to solve this if not by creating another job that merges.

Task instead of Job, but yes, this is the only way I see as well.

Reply:
> Sorry, when you criticize you should always give constructive critique and show ways for improvements

Thanks for the moral lecture. But mine was not a critique, but a lack of knowledge of how I should do that.

> I think that is also how its done in other Jobs, they write subfiles in the work folder and then have a merge task

@Atticus1806 Thanks for the suggestion, will do that.